packages/trip-details/src/index.tsx

```diff
@@ -0,0 +1,232 @@
+
+import * as Styled from "./styled";
+import TripDetail from "./trip-detail";
+import { CaloriesDetailsProps, TripDetailsProps } from "./types";
+
+// Load the default messages.
+import defaultMessages from "../i18n/en-US.yml";
+
+const MESSAGE_ID_PREFIX = "otpUi.TripDetails.";
+const messageIds = {
+  calories: `${MESSAGE_ID_PREFIX}calories`,
+  caloriesDescription: `${MESSAGE_ID_PREFIX}caloriesDescription`,
+  departure: `${MESSAGE_ID_PREFIX}departure`,
+  title: `${MESSAGE_ID_PREFIX}title`,
+  tncFare: `${MESSAGE_ID_PREFIX}tncFare`,
+  transitFare: `${MESSAGE_ID_PREFIX}transitFare`
+};
+
+/**
+ * Format text bold (used with FormattedMessage).
+ */
+// TODO: Find a better place for this utility.
+function BoldText(contents) {
+  return <b>{contents}</b>;
+}
+
+/**
+ * Default rendering if no component is provided for the CaloriesDetails
+ * slot in the TripDetails component.
+ */
+function DefaultCaloriesDetails({
+  bikeSeconds,
+  calories,
+  walkSeconds
+}: CaloriesDetailsProps): React.Element {
+  return (
+    <FormattedMessage
+      defaultMessage={defaultMessages[messageIds.caloriesDescription]}
+      id={messageIds.caloriesDescription}
+      values={{
+        // eslint-disable-next-line react/display-name
+        a: contents => (
+          <a
+            href="https://health.gov/dietaryguidelines/dga2005/document/html/chapter3.htm#table4"
+            rel="noopener noreferrer"
+            target="_blank"
+          >
+            {contents}
+          </a>
+        ),
+        b: BoldText,
+        bikeMinutes: Math.round(bikeSeconds / 60),
+        calories: Math.round(calories),
+        walkMinutes: Math.round(walkSeconds / 60)
+      }}
+    />
+  );
+}
+
+/**
+ * Renders trip details such as departure instructions, fare amount, and calories spent.
+ */
+export function TripDetails({
+  CaloriesDetails = DefaultCaloriesDetails,
+  className = "",
+  currency = "USD",
+  DepartureDetails = null,
+  FareDetails = null,
+  itinerary
+}: TripDetailsProps): ReactElement {
+  let companies;
+  itinerary.legs.forEach(leg => {
+    if (leg.tncData) {
+      companies = leg.tncData.company;
+    }
+  });
+
+  // process the transit fare
+  const fareResult = coreUtils.itinerary.calculateFares(itinerary);
+  const { maxTNCFare, minTNCFare, transitFare } = fareResult;
+  let fare;
+  if (transitFare || minTNCFare) {
+    fare = (
+      <Styled.Fare>
+        {transitFare && (
+          <Styled.TransitFare>
+            <FormattedMessage
+              defaultMessage={defaultMessages[messageIds.transitFare]}
+              id={messageIds.transitFare}
+              values={{
+                b: BoldText,
+                transitFare: (
+                  <FormattedNumber
+                    currency={currency}
+                    value={transitFare / 100}
+                    // eslint-disable-next-line react/style-prop-object
+                    style="currency"
```

Comment:
Gosh, this seems like a poor prop choice on the part of react-intl. Is there an alias we can use?